File: app/src/main/java/com/aquaguard/domain/repository/DeviceRepository.java
package com.aquaguard.domain.repository;

import androidx.lifecycle.LiveData;
import com.aquaguard.domain.model.Device;
import java.util.List;

public interface DeviceRepository {
    LiveData<List<Device>> getDevices();
    void registerDevice(String deviceId, String deviceName, RepositoryCallback<Void> callback);
    void unregisterDevice(String deviceId, RepositoryCallback<Void> callback);
    void updateDeviceName(String deviceId, String name, RepositoryCallback<Void> callback);
}

File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/domain/repository/DeviceRepository.kt
package com.aquaguard.domain.repository

import com.aquaguard.domain.model.Device
import kotlinx.coroutines.flow.Flow

interface DeviceRepository {
    fun getDevices(): Flow<List<Device>>
    suspend fun registerDevice(deviceId: String, deviceName: String): Result<Unit>
    suspend fun unregisterDevice(deviceId: String): Result<Unit>
    suspend fun updateDeviceName(deviceId: String, name: String): Result<Unit>
}
